cmd/net-crd-installer/utils/util.go

```diff
@@ -0,0 +1,156 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// CRDInstallerLabelKey is the label key used to indicate that a CRD is managed by the installer.
+	CRDInstallerLabelKey = "crd-installer.azurefleet.io/managed"
+	// AzureManagedLabelKey is the label key used to indicate that a CRD is managed by an azure resource.
+	AzureManagedLabelKey = "kubernetes.azure.com/managedby"
+	// FleetLabelValue is the value for the AzureManagedLabelKey indicating management by Fleet.
+	FleetLabelValue = "fleet"
+)
+
+var (
+	// memberIncludedCRDs defines CRDs that should be included in member clusters
+	memberIncludedCRDs = map[string]bool{
+		"serviceexports.networking.fleet.azure.com":       true,
+		"serviceimports.networking.fleet.azure.com":       true,
+		"multiclusterservices.networking.fleet.azure.com": true,
+	}
+)
+
+// InstallCRD creates/updates a Custom Resource Definition (CRD) from the provided CRD object.
+func InstallCRD(ctx context.Context, client client.Client, crd *apiextensionsv1.CustomResourceDefinition, isE2ETest bool) error {
+	klog.V(2).Infof("Installing CRD: %s", crd.Name)
+
+	existingCRD := apiextensionsv1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: crd.Name,
+		},
+	}
+
+	createOrUpdateRes, err := controllerutil.CreateOrUpdate(ctx, client, &existingCRD, func() error {
+		// Copy spec from our decoded CRD to the object we're creating/updating.
+		existingCRD.Spec = crd.Spec
+
+		// Add an additional ownership label to indicate this CRD is managed by the installer.
+		if existingCRD.Labels == nil {
+			existingCRD.Labels = make(map[string]string)
+		}
+		// Ensure the label for management by the installer is set.
+		existingCRD.Labels[CRDInstallerLabelKey] = "true"
+		// Only set the Azure managed label if this is not an E2E test.
+		// This label is needed for clean up of CRD by kube-addon-manager in production,
+		// but can interfere with E2E test as CRDs are removed from AKS clusters which
+		// are actually not hub/member clusters.
+		if !isE2ETest {
+			existingCRD.Labels[AzureManagedLabelKey] = FleetLabelValue
+		}
+		return nil
+	})
+
+	if err != nil {
+		klog.ErrorS(err, "Failed to create or update CRD", "name", crd.Name, "operation", createOrUpdateRes)
+		return err
+	}
+
+	klog.Infof("Successfully created/updated CRD %s", crd.Name)
+	return nil
+}
+
+// CollectCRDs collects CRDs from the specified path based on the mode either hub/member.
+func CollectCRDs(crdDirectoryPath, mode string, scheme *runtime.Scheme) ([]apiextensionsv1.CustomResourceDefinition, error) {
```

Comment:
Network repo import fleet repo, why copy paste the code here?

Reply:
The function signature for InstallCRD and definition for CollectCRDs has changes for this repo (We can address this in follow up PRs). Reusing GetCRDFromPath from fleet repo.

Reply:
sure, we can use a follow up PR.